quickjs.c

```diff
@@ -53043,6 +53043,33 @@ JSValue JS_NewArrayBufferCopy(JSContext *ctx, const uint8_t *buf, size_t len)
                                         TRUE);
 }
 
+/* return -1 if exception (proxy case) or TRUE/FALSE */
+int JS_IsArrayBuffer(JSContext *ctx, JSValueConst val)
+{
+    JSObject *p;
+
+    if (JS_VALUE_GET_TAG(val) == JS_TAG_OBJECT) {
+        p = JS_VALUE_GET_OBJ(val);
+        if (unlikely(p->class_id == JS_CLASS_PROXY)) {
+            JSProxyData *s = JS_GetOpaque(val, JS_CLASS_PROXY);
```

Comment:
We use the same piece of code in IsArray. I wonder if we could extract it out to some JS_IsProxy helper.

Reply:
I shall post a generic method for this and modify `JS_IsArray` accordingly.  The PR will need to be simplified accordingly.